hybrasyl/Messaging/Stats.cs

```diff
@@ -0,0 +1,312 @@
+﻿using Hybrasyl.Creatures;
+using Hybrasyl.Enums;
+using Hybrasyl.Nations;
+using Hybrasyl.Objects;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Hybrasyl.Messaging
+{
+    class HpCommand : ChatCommand
+    {
+        public new static string Command = "hp";
+        public new static string ArgumentText = "<uint hp>";
+        public new static string HelpText = "Set current HP to the specified uint value.";
+        public new static bool Privileged = false;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+            if (uint.TryParse(args[0], out uint hp))
+            {
+                user.Stats.Hp = hp;
+                user.UpdateAttributes(StatUpdateFlags.Full);
+                return Success($"HP now {hp}");
+            }
+            return Fail("The value you specified could not be parsed (uint).");
+        }
+    }
+
+    class BaseHpCommand : ChatCommand
+    {
+        public new static string Command = "basehp";
+        public new static string ArgumentText = "<uint hp>";
+        public new static string HelpText = "Set base HP to the specified uint value.";
+        public new static bool Privileged = false;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+            if (uint.TryParse(args[0], out uint hp))
+            {
+                user.Stats.BaseHp = hp;
+                user.UpdateAttributes(StatUpdateFlags.Full);
+                return Success($"Base HP now {hp}");
+            }
+            return Fail("The value you specified could not be parsed (uint)");
+        }
+    }
+
+    class MpCommand : ChatCommand
+    {
+        public new static string Command = "mp";
+        public new static string ArgumentText = "<uint mp>";
+        public new static string HelpText = "Set current HP to the specified uint value.";
+        public new static bool Privileged = false;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+            if (uint.TryParse(args[0], out uint mp))
```

Comment:
IndexOutOfRangeException if argument was not passed

Reply:
fixed